doc/adr/0012-organize-feature-specific-things-by-page-in-frontend.md

```diff
@@ -0,0 +1,567 @@
+# 12. Organize feature specific things by page in frontend
+
+Date: 202x-xx-xx
+
+## Status
+
+Proposal
+
+## Context
+
+Our project grows and both the `components` and `composables` folders grow further and further. With this it becomes
+harder to find things related to specific features.
+
+Currently, our folder structure organizes things exclusively by its type and with no further (agreed upon) organization:
+
+```
+src
+├── assets
+├── components
+│  └── controls
+├── composables
+├── lib
+├── services
+├── types
+└── views
+```
+
+With this structure it is also not clear which things (components, composables, ...) are developed to be shared between
+multiple places and which are one-of solutions for a specific feature. Also almost every feature needs to touch many
+different places in the code.
+We therefore, want to increase the colocation of things related to the same feature and make clear which things are
+shared and used by multiple features.
+
+At the moment all our big features have their own route[^1] in our application.
+
+## Decision
+
+We use the existing `components` and `composables` folders for shared components and composables.
+
+The `views` folder gets restructured to also contain things that are specific to the specific route of that view.
+
+The view folders are organized by the url paths of the view.
```

Comment:
Comment: In that case, should we revisit again the decision to use folder-based routing? Last time I checked I was not super convinced of any of the existing solutions, but maybe things have changed in the meantime.

Reply:
I'm was also not convinced by them last time I checked, especially if the path parameters get a bit more complex. And with the elis in the paths I do not think that it would work for us at all.